specification/types.md

```diff
@@ -72,6 +72,28 @@ A set of pre-defined reasons is enumerated below:
 | STALE           | The resolved value is non-authoritative or possibly out of date                                                                  |
 | ERROR           | The resolved value was the result of an error.                                                                                   |
 
+> [!NOTE]
+> The `reason` should not be limited to the reasons enumerated above. It can be any of the pre-defined reasons, or
+> any string value. Some type systems have features which can increase the ergonomics of `reason`, for instance a union
+of pre-defined types with a string, or a rust-style enumeration which allows for enumerated values to have associated
+content.
+
+```rust
```

Comment:
I am not sure if an example here is overkill.

Reply:
The solution to this varies widely language to language, so maybe just pointing it out is enough. 

The js-sdk:
```typescript
export type ResolutionReason = keyof typeof StandardResolutionReasons | (string & Record<never, never>);
```

The solution added to the python SDK.
```python
reason: typing.Optional[typing.Union[str, Reason]] = None
```

Reply:
I like having a rust example. and this is clear to me.